src/components/HeaderGap/index.tsx

```diff
@@ -0,0 +1,11 @@
+import {ReactNode} from 'react';
+import HeaderGapProps from './types';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function HeaderGap({styles}: HeaderGapProps): ReactNode {
```

Comment:
We usually don't add return type to components

Reply:
Without it the return type would be `null`, that's why it was initially typed like this:
```ts
const HeaderGap: HeaderGapComponent = () => null;
```

Reply:
@VickyStash I think we need to adjust [SplashScreenHider](https://github.com/Expensify/App/blob/main/src/components/SplashScreenHider/index.tsx) return type. Default implementation `index.tsx` returns `null`, but native implementation `index.native.tsx` returns an `Element`. The problem is that typescript infers the type from default implementation so whenever you use this component it's treated like it returns `null` which isn't entirely true..
![image](https://github.com/Expensify/App/assets/46095609/3576ad6b-dca6-4c12-8a79-2870f7ca1c15)

> And the component seems to work as expected.

It does, but the type is misleading 😅 

> But if you think it's better to keep your initial typing I'm okay with that!

I think the most clean is to make it an exception and leave the return type for `HeaderGap` and `SplashScreenHider`, as it is now

Reply:
@SzymczakJ Let's move the return type to `types.ts` and reuse this type in implementations. Go ahead and adjust [SplashScreenHider](https://github.com/Expensify/App/blob/main/src/components/SplashScreenHider/index.tsx) as well, thanks!

Reply:
Fixed comments! @VickyStash @blazejkustra what do you think?